backend/src/models/user.ts

```diff
@@ -11,12 +11,17 @@ class User extends Model<InferAttributes<User>, InferCreationAttributes<User>> {
   declare createdAt: CreationOptional<Date>;
   declare updatedAt: CreationOptional<Date>;
 
-  isValidPassword(password: string) {
-    return bcrypt.compare(password, this.password);
+  async isValidPassword(password: string) {
+    return await bcrypt.compare(password, this.password);
+  }
+
+  async encryptPassword(password: string) {
+    const hash = await bcrypt.hash(password, parseInt(process.env.SALT_ROUNDS || '10'));
+    return hash;
   }
 
   generateToken() {
-    const body = { _id: this.id, email: this.email };
+    const body = { id: this.id, email: this.email };
     const token = jwt.sign({ user: body }, process.env.JWT_SECRET_KEY || '');
     return token;
```

Comment:
Same nitpick as above 😄

Reply:
We missed this. I usually prefer this way because in my opinion it's more readable but it's a short function and function name says what it does so it can be shortened. Fixed 😂